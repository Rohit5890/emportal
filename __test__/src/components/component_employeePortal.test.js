import React from 'react';
import ReactDOM from 'react-dom';
import {configure, shallow } from 'enzyme';
import Adapter from "enzyme-adapter-react-16";
import EmployeePortal from '../../../src/components/component_employeePortal';

import { Dropdown, DefaultButton} from 'office-ui-fabric-react';
import { setTimeout } from 'timers';

configure({ adapter: new Adapter() });

let data  =  {"id":1,"first_name":"George","last_name":"Bluth","avatar":"https://s3.amazonaws.com/uifaces/faces/twitter/calebogden/128.jpg"};

const component =  shallow(<EmployeePortal  empData={data}/>);

test('EmployeePortal', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<EmployeePortal/>, div);
    });
}),

test('EmployeePortal snapshot rendering',()=>{
    expect(component).toMatchSnapshot()
}),

test('Form element is present ', ()=>{
    expect(component.find('.flex-row > form').length).toEqual(1)
});

test('All Input elemnets are rendered', ()=>{
    expect(component.find(Dropdown).length).toEqual(2)
});


test('Component expects empData prop',()=>{
    expect(component.instance().props.empData.id).toBe(1);
})