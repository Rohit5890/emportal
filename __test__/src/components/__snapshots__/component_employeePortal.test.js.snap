// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmployeePortal snapshot rendering 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EmployeePortal
    empData={
        Object {
            "avatar": "https://s3.amazonaws.com/uifaces/faces/twitter/calebogden/128.jpg",
            "first_name": "George",
            "id": 1,
            "last_name": "Bluth",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="flex-row"
>
        <form
                onReset={[Function]}
                onSubmit={[Function]}
        >
                <StyledWithResponsiveMode
                        label="Departments:"
                        onChanged={[Function]}
                        options={
                                Array [
                                        Object {
                                          "key": "_hr",
                                          "stateKey": "deptKey",
                                          "text": "HR",
                                        },
                                        Object {
                                          "key": "_eng",
                                          "stateKey": "deptKey",
                                          "text": "Engineering",
                                        },
                                      ]
                        }
                        placeHolder="Select an Department"
                        selectedKey=""
                />
                <StyledWithResponsiveMode
                        label="Employee Id:"
                        onBlur=""
                        onChanged={[Function]}
                        onFocus=""
                        options={undefined}
                        placeHolder="Select an Department"
                        selectedKey=""
                />
                <CustomizedDefaultButton
                        allowDisabledFocus={true}
                        checked="checked"
                        className="submit"
                        data-automation-id="test"
                        primary={true}
                        text="Get Details"
                        type="submit"
                />
                <CustomizedDefaultButton
                        allowDisabledFocus={true}
                        checked="checked"
                        className="reset"
                        data-automation-id="test_one"
                        primary={true}
                        text="Clear"
                        type="reset"
                />
        </form>
        <div
                className="details-wrap"
        >
                
        </div>
</div>,
      "className": "wrapper",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <form
            onReset={[Function]}
            onSubmit={[Function]}
>
            <StyledWithResponsiveMode
                        label="Departments:"
                        onChanged={[Function]}
                        options={
                                    Array [
                                                Object {
                                                  "key": "_hr",
                                                  "stateKey": "deptKey",
                                                  "text": "HR",
                                                },
                                                Object {
                                                  "key": "_eng",
                                                  "stateKey": "deptKey",
                                                  "text": "Engineering",
                                                },
                                              ]
                        }
                        placeHolder="Select an Department"
                        selectedKey=""
            />
            <StyledWithResponsiveMode
                        label="Employee Id:"
                        onBlur=""
                        onChanged={[Function]}
                        onFocus=""
                        options={undefined}
                        placeHolder="Select an Department"
                        selectedKey=""
            />
            <CustomizedDefaultButton
                        allowDisabledFocus={true}
                        checked="checked"
                        className="submit"
                        data-automation-id="test"
                        primary={true}
                        text="Get Details"
                        type="submit"
            />
            <CustomizedDefaultButton
                        allowDisabledFocus={true}
                        checked="checked"
                        className="reset"
                        data-automation-id="test_one"
                        primary={true}
                        text="Clear"
                        type="reset"
            />
</form>,
          <div
            className="details-wrap"
>
            
</div>,
        ],
        "className": "flex-row",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <StyledWithResponsiveMode
                label="Departments:"
                onChanged={[Function]}
                options={
                                Array [
                                                Object {
                                                  "key": "_hr",
                                                  "stateKey": "deptKey",
                                                  "text": "HR",
                                                },
                                                Object {
                                                  "key": "_eng",
                                                  "stateKey": "deptKey",
                                                  "text": "Engineering",
                                                },
                                              ]
                }
                placeHolder="Select an Department"
                selectedKey=""
/>,
              <StyledWithResponsiveMode
                label="Employee Id:"
                onBlur=""
                onChanged={[Function]}
                onFocus=""
                options={undefined}
                placeHolder="Select an Department"
                selectedKey=""
/>,
              <CustomizedDefaultButton
                allowDisabledFocus={true}
                checked="checked"
                className="submit"
                data-automation-id="test"
                primary={true}
                text="Get Details"
                type="submit"
/>,
              <CustomizedDefaultButton
                allowDisabledFocus={true}
                checked="checked"
                className="reset"
                data-automation-id="test_one"
                primary={true}
                text="Clear"
                type="reset"
/>,
            ],
            "onReset": [Function],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "label": "Departments:",
                "onChanged": [Function],
                "options": Array [
                  Object {
                    "key": "_hr",
                    "stateKey": "deptKey",
                    "text": "HR",
                  },
                  Object {
                    "key": "_eng",
                    "stateKey": "deptKey",
                    "text": "Engineering",
                  },
                ],
                "placeHolder": "Select an Department",
                "selectedKey": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "label": "Employee Id:",
                "onBlur": "",
                "onChanged": [Function],
                "onFocus": "",
                "options": undefined,
                "placeHolder": "Select an Department",
                "selectedKey": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "allowDisabledFocus": true,
                "checked": "checked",
                "className": "submit",
                "data-automation-id": "test",
                "primary": true,
                "text": "Get Details",
                "type": "submit",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "allowDisabledFocus": true,
                "checked": "checked",
                "className": "reset",
                "data-automation-id": "test_one",
                "primary": true,
                "text": "Clear",
                "type": "reset",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              false,
              undefined,
            ],
            "className": "details-wrap",
          },
          "ref": null,
          "rendered": Array [
            false,
            undefined,
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="flex-row"
>
          <form
                    onReset={[Function]}
                    onSubmit={[Function]}
          >
                    <StyledWithResponsiveMode
                              label="Departments:"
                              onChanged={[Function]}
                              options={
                                        Array [
                                                  Object {
                                                    "key": "_hr",
                                                    "stateKey": "deptKey",
                                                    "text": "HR",
                                                  },
                                                  Object {
                                                    "key": "_eng",
                                                    "stateKey": "deptKey",
                                                    "text": "Engineering",
                                                  },
                                                ]
                              }
                              placeHolder="Select an Department"
                              selectedKey=""
                    />
                    <StyledWithResponsiveMode
                              label="Employee Id:"
                              onBlur=""
                              onChanged={[Function]}
                              onFocus=""
                              options={undefined}
                              placeHolder="Select an Department"
                              selectedKey=""
                    />
                    <CustomizedDefaultButton
                              allowDisabledFocus={true}
                              checked="checked"
                              className="submit"
                              data-automation-id="test"
                              primary={true}
                              text="Get Details"
                              type="submit"
                    />
                    <CustomizedDefaultButton
                              allowDisabledFocus={true}
                              checked="checked"
                              className="reset"
                              data-automation-id="test_one"
                              primary={true}
                              text="Clear"
                              type="reset"
                    />
          </form>
          <div
                    className="details-wrap"
          >
                    
          </div>
</div>,
        "className": "wrapper",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <form
              onReset={[Function]}
              onSubmit={[Function]}
>
              <StyledWithResponsiveMode
                            label="Departments:"
                            onChanged={[Function]}
                            options={
                                          Array [
                                                        Object {
                                                          "key": "_hr",
                                                          "stateKey": "deptKey",
                                                          "text": "HR",
                                                        },
                                                        Object {
                                                          "key": "_eng",
                                                          "stateKey": "deptKey",
                                                          "text": "Engineering",
                                                        },
                                                      ]
                            }
                            placeHolder="Select an Department"
                            selectedKey=""
              />
              <StyledWithResponsiveMode
                            label="Employee Id:"
                            onBlur=""
                            onChanged={[Function]}
                            onFocus=""
                            options={undefined}
                            placeHolder="Select an Department"
                            selectedKey=""
              />
              <CustomizedDefaultButton
                            allowDisabledFocus={true}
                            checked="checked"
                            className="submit"
                            data-automation-id="test"
                            primary={true}
                            text="Get Details"
                            type="submit"
              />
              <CustomizedDefaultButton
                            allowDisabledFocus={true}
                            checked="checked"
                            className="reset"
                            data-automation-id="test_one"
                            primary={true}
                            text="Clear"
                            type="reset"
              />
</form>,
            <div
              className="details-wrap"
>
              
</div>,
          ],
          "className": "flex-row",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <StyledWithResponsiveMode
                  label="Departments:"
                  onChanged={[Function]}
                  options={
                                    Array [
                                                      Object {
                                                        "key": "_hr",
                                                        "stateKey": "deptKey",
                                                        "text": "HR",
                                                      },
                                                      Object {
                                                        "key": "_eng",
                                                        "stateKey": "deptKey",
                                                        "text": "Engineering",
                                                      },
                                                    ]
                  }
                  placeHolder="Select an Department"
                  selectedKey=""
/>,
                <StyledWithResponsiveMode
                  label="Employee Id:"
                  onBlur=""
                  onChanged={[Function]}
                  onFocus=""
                  options={undefined}
                  placeHolder="Select an Department"
                  selectedKey=""
/>,
                <CustomizedDefaultButton
                  allowDisabledFocus={true}
                  checked="checked"
                  className="submit"
                  data-automation-id="test"
                  primary={true}
                  text="Get Details"
                  type="submit"
/>,
                <CustomizedDefaultButton
                  allowDisabledFocus={true}
                  checked="checked"
                  className="reset"
                  data-automation-id="test_one"
                  primary={true}
                  text="Clear"
                  type="reset"
/>,
              ],
              "onReset": [Function],
              "onSubmit": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "label": "Departments:",
                  "onChanged": [Function],
                  "options": Array [
                    Object {
                      "key": "_hr",
                      "stateKey": "deptKey",
                      "text": "HR",
                    },
                    Object {
                      "key": "_eng",
                      "stateKey": "deptKey",
                      "text": "Engineering",
                    },
                  ],
                  "placeHolder": "Select an Department",
                  "selectedKey": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "label": "Employee Id:",
                  "onBlur": "",
                  "onChanged": [Function],
                  "onFocus": "",
                  "options": undefined,
                  "placeHolder": "Select an Department",
                  "selectedKey": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "allowDisabledFocus": true,
                  "checked": "checked",
                  "className": "submit",
                  "data-automation-id": "test",
                  "primary": true,
                  "text": "Get Details",
                  "type": "submit",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "allowDisabledFocus": true,
                  "checked": "checked",
                  "className": "reset",
                  "data-automation-id": "test_one",
                  "primary": true,
                  "text": "Clear",
                  "type": "reset",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "form",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                false,
                undefined,
              ],
              "className": "details-wrap",
            },
            "ref": null,
            "rendered": Array [
              false,
              undefined,
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
